airbyte-webapp/src/views/Connection/ConnectionForm/components/ScheduleField.tsx

```diff
@@ -0,0 +1,182 @@
+import { useFrequencyDropdownData } from "../formConfig";
+import styles from "./ScheduleField.module.scss";
+
+interface ScheduleFieldProps {
+  scheduleData: ConnectionScheduleData | undefined;
+  scheduleType: ConnectionScheduleType | undefined;
+  mode: ConnectionFormMode;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  setFieldValue: (field: string, value: any, shouldValidate?: boolean | undefined) => void;
+  onDropDownSelect?: (item: DropDownRow.IDataItem) => void;
+}
+
+const ScheduleField: React.FC<ScheduleFieldProps> = ({
+  scheduleData,
+  scheduleType,
+  mode,
+  setFieldValue,
+  onDropDownSelect,
+}) => {
+  const [isCustomSchedule, setIsCustomSchedule] = useState<boolean>(scheduleType === ConnectionScheduleType.cron);
+  const { formatMessage } = useIntl();
+  const frequencies = useFrequencyDropdownData(scheduleData);
+
+  const onScheduleChange = (
+    item: DropDownRow.IDataItem,
+    field: FieldInputProps<ConnectionScheduleDataBasicSchedule>
+  ) => {
+    onDropDownSelect?.(item);
+    setFieldValue(field.name, item.value);
+
+    // Also set scheduleType for yup validation
+    const isManualOrCron = item.value === ConnectionScheduleType.manual || item.value === ConnectionScheduleType.cron;
+    const scheduleType = isManualOrCron ? (item.value as ConnectionScheduleType) : ConnectionScheduleType.basic;
+
+    setFieldValue("scheduleType", scheduleType);
+
+    // Show cron and timezone fields based on the frequency
+    if (item.value === ConnectionScheduleType.cron) {
+      setIsCustomSchedule(true);
+    } else {
+      setIsCustomSchedule(false);
+    }
+  };
+
+  const getBasicScheduleValue = (value: ConnectionScheduleDataBasicSchedule) => {
+    // To set the initial value for the frequency dropdown. Only for Manual and Custom
+    if (!value) {
+      if (scheduleType === ConnectionScheduleType.cron) {
+        return formatMessage({
+          id: "frequency.cron",
+        }).toLowerCase();
+      }
+
+      return formatMessage({
+        id: "frequency.manual",
+      }).toLowerCase();
+    }
+
+    return value;
+  };
+
+  const getZoneValue = (currentSelectedZone: string) => {
+    if (!currentSelectedZone) {
+      return cronTimeZones[0].value;
+    }
+
+    return currentSelectedZone;
+  };
+
+  const onCronInputChange = (
+    event: ChangeEvent<HTMLInputElement>,
+    field: FieldInputProps<ConnectionScheduleDataCron>
+  ) => {
+    setFieldValue(field.name, {
+      ...field.value,
+      cronExpression: event.currentTarget.value,
+    });
+  };
+
+  const onCronZoneChange = (event: DropDownRow.IDataItem, field: FieldInputProps<ConnectionScheduleDataCron>) => {
+    setFieldValue(field.name, {
+      ...field.value,
+      cronTimeZone: event.value,
+    });
+  };
+
+  const cronTimeZones = useMemo(() => {
+    return availableCronTimeZones.map((zone: string) => ({ label: zone, value: zone }));
+  }, []);
+
+  return (
+    <>
+      <Field name="scheduleData.basicSchedule">
```

Comment:
I don't think that we should use `scheduleData.basicSchedule` as the field name here, and `scheduleData.cron` as the field name below. This leads to confusing behavior, because we don't always want those fields to be set depending on the type of schedule the user selects, like I described in my comment above.

Instead, I think this component should only contain a single `<Field>`, which should have `name="scheduleData"`. This Field should contain the `Replication frequency` dropdown **as well as** the `Cron expression` inputs (conditionally) within it, rather than that being controlled by a separate Field. 

Then, we should change the `onScheduleChange()`, `onCronInputChange()`, and `onCronZoneChange()` methods to all call `setFieldValue()` with the `scheduleData` field name, and they should set the entire `scheduleData` object to the correct value (and in the manual case, this value is `undefined`).

I think this would be simpler and more reliable than trying to set properties within the `scheduleData` object through different Fields.

Reply:
Yeah I had this approach also in my mind. Looks fair to me.